### PR TITLE
Adding ActionsCluster - Issue #92

### PR DESCRIPTION
**Pull request by donnchadh**
Issue #92
Adding ActionsCluster based on actions-cluster.xml and the Matter Core Specification.

### CODE REVIEW

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+ */
+export const ActionsCluster = Cluster({
+    id: 0x25,
+    name: "Actions",
+    revision: 1,
+
+    /** @see {@link MatterCoreSpecificationV1_0} § 9.14.6 */
+    attributes: {
+        /** List of actions. */
+        actionList: Attribute(0, TlvArray(TlvAction), { default: [] }),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:289]: please add {maxLength: 256} in the TlvArray as second parameter

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+        /** List of endpoint lists. */
+        endpointLists: Attribute(1, TlvArray(TlvEndpointList), { default: [] }),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:292]: please add {maxLength: 256} in the TlvArray as second parameter

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+});
+
+/**
+ * Action error.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.3
+ */
+export const enum ActionErrorEnum {
+    Unknown = 0,
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:65]:
```suggestion
    /** Other reason not listed below */
    Unknown = 0,
```

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+
+    Interrupted = 1,
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:67]:
```suggestion
    /** The action was inter­ rupted by another com­ mand or interaction */
    Interrupted = 1,
```

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Attribute, Cluster, Event, EventPriority, OptionalAttribute, OptionalCommand, TlvNoResponse } from "./Cluster.js";
+import { TlvArray } from "../tlv/TlvArray.js";
+import { TlvString } from "../tlv/TlvString.js";
+import { TlvEnum, TlvBitmap, TlvUInt16, TlvUInt32 } from "../tlv/TlvNumber.js";
+import { TlvField, TlvObject, TlvOptionalField } from "../tlv/TlvObject.js";
+import { BitFlag } from "../schema/BitmapSchema.js";
+
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:13]:
```suggestion
import { MatterCoreSpecificationV1_0 } from "../spec/Specifications.js";

```

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+ * Provides information about endpoint conformance to a release of a device type definition.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.5
+ */
+const TlvEndpointList = TlvObject({
+    /** Indicates the device type definition */
+    endpointListId: TlvField(0, TlvUInt16),
+
+    /** Indicates the implemented revision of the device type definition */
+    name: TlvField(1, TlvString),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:166]: We should add also maxLength 32 here?

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+
+    type: TlvField(2, TlvEnum<EndpointListType>()),
+
+    endpoints: TlvField(3, TlvArray(TlvUInt16)),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:170]:
```suggestion
    /** List of endpoint numbers. */
    endpoints: TlvField(3, TlvArray(EndpointNumber)),
```

(and add import please!)

Comment (Apollon77) [reply packages/matter.js/src/cluster/ActionsCluster.ts:170]: and maxLength: 256

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.9.5
+ */
+const TlvEndpointList = TlvObject({
+    /** Indicates the device type definition */
+    endpointListId: TlvField(0, TlvUInt16),
+
+    /** Indicates the implemented revision of the device type definition */
+    name: TlvField(1, TlvString),
+
+    type: TlvField(2, TlvEnum<EndpointListType>()),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:168]:
```suggestion
    /** Indicates the type of endpoint list. */
    type: TlvField(2, TlvEnum<EndpointListType>()),
```

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+    /** Higher priority notification */
+    Alarm = 6,
+}
+
+/**
+ * Provides information about endpoint conformance to a release of a device type definition.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 9.14.6
+ */
+const TlvAction = TlvObject({
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:123]: Adding the missing comments would be cool

File: packages/matter.js/src/cluster/ActionsCluster.ts [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+    /** Indicates the device type definition */
+    actionID: TlvField(0, TlvUInt16),
+
+    /** Indicates the implemented revision of the device type definition */
+    name: TlvField(1, TlvString.bound({ maxLength: 32 })),
+
+    type: TlvField(2, TlvEnum<ActionTypeEnum>()),
+
+    endpointListID: TlvField(3, TlvUInt16),
```
Comment (Apollon77) [inline packages/matter.js/src/cluster/ActionsCluster.ts:132]:
```suggestion
    endpointListId: TlvField(3, TlvUInt16),
```

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Thank you very much, some smaller things I found. Thank you for adjustments

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 1 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 2 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 4 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts:71]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 6 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 8 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts:177]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 10 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 9 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts:127]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 7 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 5 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts:15]: Done.

File: packages/matter.js/src/cluster/ActionsCluster.ts (hunk 3 above)
Comment (donnchadh) [reply packages/matter.js/src/cluster/ActionsCluster.ts:68]: Done

**Review by Apollon77 — APPROVED** [review]
Thank you, approved

**Pull request merged**